ToolBox/Benzinga/BenzingaNewsDataConverter.cs

```diff
@@ -0,0 +1,339 @@
+        public BenzingaNewsDataConverter(DirectoryInfo sourceDirectory, DirectoryInfo destinationDirectory, DirectoryInfo processedFilesDirectory)
+        {
+            _sourceDirectory = sourceDirectory;
+            _destinationDirectory = destinationDirectory;
+            _processedFilesDirectory = processedFilesDirectory;
+            _destinationDirectory.Create();
+            _isWindows = OS.IsWindows;
+
+            _mapFileProvider = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"));
+            _mapFileResolver = _mapFileProvider.Get(Market.USA);
+        }
+
+        public bool Convert(DateTime date)
+        {
+            var newsArticles = new DirectoryInfo(Path.Combine(
+                _sourceDirectory.FullName,
+                $"{date.Year}",
+                date.Month.ToStringInvariant().PadLeft(2, '0'),
+                date.Day.ToStringInvariant().PadLeft(2, '0')));
+
+            if (!newsArticles.Exists)
+            {
+                Log.Error($"BenzingaDataConverter.Convert(): Data not found for date {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            WriteToFile(Process(newsArticles), date);
+            return true;
+        }
+
+        /// <summary>
+        /// Process the data to BenzingaNews instances. This will process both historical and API retrieved news data.
+        /// </summary>
+        /// <param name="dateDirectory">Directory for a given date, e.g. ./root/2018/12/27/</param>
+        /// <returns></returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
+            {
+                var document = new XmlDocument();
+                document.LoadXml(File.ReadAllText(publication.FullName));
+
+                var news = BenzingaNewsFactory.CreateBenzingaNewsFromRSS(JsonConvert.SerializeXmlNode(document), _mapFileProvider, _mapFileResolver);
+                if (news == null)
+                {
+                    continue;
+                }
+
+                yield return news;
+            }
+
+            foreach (var publication in dateDirectory.GetFiles("*.json", SearchOption.TopDirectoryOnly))
+            {
+                var news = BenzingaNewsFactory.CreateBenzingaNewsFromJSON(File.ReadAllText(publication.FullName), _mapFileResolver);
+                foreach (var article in news)
+                {
+                    yield return article;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
```

Comment:
Please split this method into logical parts.

In any case, almost 200 LOC for writing data is suspicious.